CLOUDSTACK-8239 Add VirtIO SCSI support for KVM hosts

nathanejohnson opened the pull request:
This is an initial implementation that we have working in the lab.

![devmanagerscreenshot](https://cloud.githubusercontent.com/assets/1360240/23133419/981f89ce-f756-11e6-8d17-9ddae264fca8.png)
![spanned_sceenshot](https://cloud.githubusercontent.com/assets/1360240/23133418/981e55cc-f756-11e6-9fd7-f7757c82438c.png)


[win_virtio_scsi.xml.zip](https://github.com/apache/cloudstack/files/788014/win_virtio_scsi.xml.zip)

@wido @syed Would love to get some feedback on this approach.

kiwiflyer commented:
@remibergsma @ustcweizhou This might be of interest to you guys as well.

remibergsma commented:
Nice work @kiwiflyer @dmabry, thanks! Will give it a go soon.

wido commented:
Very nice indeed! I will take a look asap.

dmabry commented:
We are deploying this to our QA environment right now and hope to have it tested in a few days.  Great work @kiwiflyer and @nathanejohnson.

borisstoyanov commented:
Thanks for this PR @nathanejohnson, I will build it and run smoketests agains it. 
@blueorangutan package

blueorangutan commented:
@borisstoyanov a Jenkins job has been kicked to build packages. I'll keep you posted as I make progress.

blueorangutan commented:
Packaging result: ✔centos6 ✔centos7 ✔debian. JID-522

borisstoyanov commented:
@blueorangutan test

blueorangutan commented:
@borisstoyanov a Trillian-Jenkins test job (centos7 mgmt + kvm-centos7) has been kicked to run smoke tests

blueorangutan commented:
<b>Trillian test result (tid-878)</b>
Environment: kvm-centos7 (x2), Advanced Networking with Mgmt server 7
Total time taken: 32944 seconds
Marvin logs: https://github.com/blueorangutan/acs-prs/releases/download/trillian/pr1955-t878-kvm-centos7.zip
Intermitten failure detected: /marvin/tests/smoke/test_privategw_acl.py
Intermitten failure detected: /marvin/tests/smoke/test_snapshots.py
Intermitten failure detected: /marvin/tests/smoke/test_vpc_redundant.py
Test completed. 47 look ok, 2 have error(s)


Test | Result | Time (s) | Test File
--- | --- | --- | ---
test_04_rvpc_privategw_static_routes | `Failure` | 329.25 | test_privategw_acl.py
test_02_list_snapshots_with_removed_data_store | `Error` | 0.03 | test_snapshots.py
test_01_vpc_site2site_vpn | Success | 164.34 | test_vpc_vpn.py
test_01_vpc_remote_access_vpn | Success | 70.82 | test_vpc_vpn.py
test_01_redundant_vpc_site2site_vpn | Success | 270.12 | test_vpc_vpn.py
test_02_VPC_default_routes | Success | 288.94 | test_vpc_router_nics.py
test_01_VPC_nics_after_destroy | Success | 511.96 | test_vpc_router_nics.py
test_05_rvpc_multi_tiers | Success | 515.21 | test_vpc_redundant.py
test_04_rvpc_network_garbage_collector_nics | Success | 1413.54 | test_vpc_redundant.py
test_03_create_redundant_VPC_1tier_2VMs_2IPs_2PF_ACL_reboot_routers | Success | 557.28 | test_vpc_redundant.py
test_02_redundant_VPC_default_routes | Success | 743.95 | test_vpc_redundant.py
test_01_create_redundant_VPC_2tiers_4VMs_4IPs_4PF_ACL | Success | 1282.32 | test_vpc_redundant.py
test_09_delete_detached_volume | Success | 156.30 | test_volumes.py
test_08_resize_volume | Success | 151.31 | test_volumes.py
test_07_resize_fail | Success | 161.21 | test_volumes.py
test_06_download_detached_volume | Success | 156.04 | test_volumes.py
test_05_detach_volume | Success | 155.69 | test_volumes.py
test_04_delete_attached_volume | Success | 151.00 | test_volumes.py
test_03_download_attached_volume | Success | 156.04 | test_volumes.py
test_02_attach_volume | Success | 124.19 | test_volumes.py
test_01_create_volume | Success | 724.31 | test_volumes.py
test_03_delete_vm_snapshots | Success | 275.70 | test_vm_snapshots.py
test_02_revert_vm_snapshots | Success | 95.60 | test_vm_snapshots.py
test_01_create_vm_snapshots | Success | 166.79 | test_vm_snapshots.py
test_deploy_vm_multiple | Success | 257.28 | test_vm_life_cycle.py
test_deploy_vm | Success | 0.02 | test_vm_life_cycle.py
test_advZoneVirtualRouter | Success | 0.02 | test_vm_life_cycle.py
test_10_attachAndDetach_iso | Success | 26.57 | test_vm_life_cycle.py
test_09_expunge_vm | Success | 125.21 | test_vm_life_cycle.py
test_08_migrate_vm | Success | 41.07 | test_vm_life_cycle.py
test_07_restore_vm | Success | 0.10 | test_vm_life_cycle.py
test_06_destroy_vm | Success | 125.79 | test_vm_life_cycle.py
test_03_reboot_vm | Success | 125.68 | test_vm_life_cycle.py
test_02_start_vm | Success | 10.13 | test_vm_life_cycle.py
test_01_stop_vm | Success | 40.37 | test_vm_life_cycle.py
test_CreateTemplateWithDuplicateName | Success | 60.62 | test_templates.py
test_08_list_system_templates | Success | 0.02 | test_templates.py
test_07_list_public_templates | Success | 0.03 | test_templates.py
test_05_template_permissions | Success | 0.04 | test_templates.py
test_04_extract_template | Success | 5.13 | test_templates.py
test_03_delete_template | Success | 5.08 | test_templates.py
test_02_edit_template | Success | 90.14 | test_templates.py
test_01_create_template | Success | 25.28 | test_templates.py
test_10_destroy_cpvm | Success | 161.61 | test_ssvm.py
test_09_destroy_ssvm | Success | 163.57 | test_ssvm.py
test_08_reboot_cpvm | Success | 101.43 | test_ssvm.py
test_07_reboot_ssvm | Success | 133.41 | test_ssvm.py
test_06_stop_cpvm | Success | 131.63 | test_ssvm.py
test_05_stop_ssvm | Success | 134.59 | test_ssvm.py
test_04_cpvm_internals | Success | 1.17 | test_ssvm.py
test_03_ssvm_internals | Success | 3.70 | test_ssvm.py
test_02_list_cpvm_vm | Success | 0.09 | test_ssvm.py
test_01_list_sec_storage_vm | Success | 0.09 | test_ssvm.py
test_01_snapshot_root_disk | Success | 10.93 | test_snapshots.py
test_04_change_offering_small | Success | 239.36 | test_service_offerings.py
test_03_delete_service_offering | Success | 0.06 | test_service_offerings.py
test_02_edit_service_offering | Success | 0.04 | test_service_offerings.py
test_01_create_service_offering | Success | 0.09 | test_service_offerings.py
test_02_sys_template_ready | Success | 0.09 | test_secondary_storage.py
test_01_sys_vm_start | Success | 0.12 | test_secondary_storage.py
test_09_reboot_router | Success | 35.26 | test_routers.py
test_08_start_router | Success | 30.22 | test_routers.py
test_07_stop_router | Success | 10.12 | test_routers.py
test_06_router_advanced | Success | 0.07 | test_routers.py
test_05_router_basic | Success | 0.03 | test_routers.py
test_04_restart_network_wo_cleanup | Success | 5.68 | test_routers.py
test_03_restart_network_cleanup | Success | 60.39 | test_routers.py
test_02_router_internal_adv | Success | 1.11 | test_routers.py
test_01_router_internal_basic | Success | 0.60 | test_routers.py
test_router_dns_guestipquery | Success | 76.68 | test_router_dns.py
test_router_dns_externalipquery | Success | 0.06 | test_router_dns.py
test_router_dhcphosts | Success | 277.57 | test_router_dhcphosts.py
test_router_dhcp_opts | Success | 21.63 | test_router_dhcphosts.py
test_01_updatevolumedetail | Success | 0.05 | test_resource_detail.py
test_01_reset_vm_on_reboot | Success | 145.78 | test_reset_vm_on_reboot.py
test_createRegion | Success | 0.03 | test_regions.py
test_create_pvlan_network | Success | 5.16 | test_pvlan.py
test_dedicatePublicIpRange | Success | 0.30 | test_public_ip_range.py
test_03_vpc_privategw_restart_vpc_cleanup | Success | 473.41 | test_privategw_acl.py
test_02_vpc_privategw_static_routes | Success | 369.24 | test_privategw_acl.py
test_01_vpc_privategw_acl | Success | 91.58 | test_privategw_acl.py
test_01_primary_storage_nfs | Success | 35.77 | test_primary_storage.py
test_createPortablePublicIPRange | Success | 15.26 | test_portable_publicip.py
test_createPortablePublicIPAcquire | Success | 15.31 | test_portable_publicip.py
test_isolate_network_password_server | Success | 89.39 | test_password_server.py
test_UpdateStorageOverProvisioningFactor | Success | 0.16 | test_over_provisioning.py
test_oobm_zchange_password | Success | 30.57 | test_outofbandmanagement.py
test_oobm_multiple_mgmt_server_ownership | Success | 16.28 | test_outofbandmanagement.py
test_oobm_issue_power_status | Success | 10.25 | test_outofbandmanagement.py
test_oobm_issue_power_soft | Success | 15.30 | test_outofbandmanagement.py
test_oobm_issue_power_reset | Success | 15.25 | test_outofbandmanagement.py
test_oobm_issue_power_on | Success | 15.24 | test_outofbandmanagement.py
test_oobm_issue_power_off | Success | 15.34 | test_outofbandmanagement.py
test_oobm_issue_power_cycle | Success | 15.34 | test_outofbandmanagement.py
test_oobm_enabledisable_across_clusterzones | Success | 87.36 | test_outofbandmanagement.py
test_oobm_enable_feature_valid | Success | 0.12 | test_outofbandmanagement.py
test_oobm_enable_feature_invalid | Success | 0.11 | test_outofbandmanagement.py
test_oobm_disable_feature_valid | Success | 5.23 | test_outofbandmanagement.py
test_oobm_disable_feature_invalid | Success | 0.30 | test_outofbandmanagement.py
test_oobm_configure_invalid_driver | Success | 0.12 | test_outofbandmanagement.py
test_oobm_configure_default_driver | Success | 0.19 | test_outofbandmanagement.py
test_oobm_background_powerstate_sync | Success | 18.38 | test_outofbandmanagement.py
test_extendPhysicalNetworkVlan | Success | 15.53 | test_non_contigiousvlan.py
test_01_nic | Success | 438.67 | test_nic.py
test_releaseIP | Success | 162.20 | test_network.py
test_reboot_router | Success | 403.23 | test_network.py
test_public_ip_user_account | Success | 10.19 | test_network.py
test_public_ip_admin_account | Success | 40.22 | test_network.py
test_network_rules_acquired_public_ip_3_Load_Balancer_Rule | Success | 66.78 | test_network.py
test_network_rules_acquired_public_ip_2_nat_rule | Success | 61.67 | test_network.py
test_network_rules_acquired_public_ip_1_static_nat_rule | Success | 124.08 | test_network.py
test_delete_account | Success | 287.22 | test_network.py
test_02_port_fwd_on_non_src_nat | Success | 55.52 | test_network.py
test_01_port_fwd_on_src_nat | Success | 111.62 | test_network.py
test_nic_secondaryip_add_remove | Success | 201.82 | test_multipleips_per_nic.py
login_test_saml_user | Success | 17.84 | test_login.py
test_assign_and_removal_lb | Success | 133.27 | test_loadbalance.py
test_02_create_lb_rule_non_nat | Success | 187.08 | test_loadbalance.py
test_01_create_lb_rule_src_nat | Success | 227.66 | test_loadbalance.py
test_03_list_snapshots | Success | 0.04 | test_list_ids_parameter.py
test_02_list_templates | Success | 0.03 | test_list_ids_parameter.py
test_01_list_volumes | Success | 0.02 | test_list_ids_parameter.py
test_07_list_default_iso | Success | 0.04 | test_iso.py
test_05_iso_permissions | Success | 0.04 | test_iso.py
test_04_extract_Iso | Success | 5.12 | test_iso.py
test_03_delete_iso | Success | 95.12 | test_iso.py
test_02_edit_iso | Success | 0.04 | test_iso.py
test_01_create_iso | Success | 20.68 | test_iso.py
test_04_rvpc_internallb_haproxy_stats_on_all_interfaces | Success | 192.96 | test_internal_lb.py
test_03_vpc_internallb_haproxy_stats_on_all_interfaces | Success | 147.77 | test_internal_lb.py
test_02_internallb_roundrobin_1RVPC_3VM_HTTP_port80 | Success | 514.39 | test_internal_lb.py
test_01_internallb_roundrobin_1VPC_3VM_HTTP_port80 | Success | 428.47 | test_internal_lb.py
test_dedicateGuestVlanRange | Success | 10.20 | test_guest_vlan_range.py
test_UpdateConfigParamWithScope | Success | 0.09 | test_global_settings.py
test_rolepermission_lifecycle_update | Success | 5.96 | test_dynamicroles.py
test_rolepermission_lifecycle_list | Success | 5.66 | test_dynamicroles.py
test_rolepermission_lifecycle_delete | Success | 5.58 | test_dynamicroles.py
test_rolepermission_lifecycle_create | Success | 5.59 | test_dynamicroles.py
test_rolepermission_lifecycle_concurrent_updates | Success | 5.67 | test_dynamicroles.py
test_role_lifecycle_update_role_inuse | Success | 5.61 | test_dynamicroles.py
test_role_lifecycle_update | Success | 10.66 | test_dynamicroles.py
test_role_lifecycle_list | Success | 5.60 | test_dynamicroles.py
test_role_lifecycle_delete | Success | 5.60 | test_dynamicroles.py
test_role_lifecycle_create | Success | 5.60 | test_dynamicroles.py
test_role_inuse_deletion | Success | 5.60 | test_dynamicroles.py
test_role_account_acls_multiple_mgmt_servers | Success | 7.12 | test_dynamicroles.py
test_role_account_acls | Success | 7.01 | test_dynamicroles.py
test_default_role_deletion | Success | 5.66 | test_dynamicroles.py
test_04_create_fat_type_disk_offering | Success | 0.05 | test_disk_offerings.py
test_03_delete_disk_offering | Success | 0.03 | test_disk_offerings.py
test_02_edit_disk_offering | Success | 0.03 | test_disk_offerings.py
test_02_create_sparse_type_disk_offering | Success | 0.05 | test_disk_offerings.py
test_01_create_disk_offering | Success | 0.08 | test_disk_offerings.py
test_deployvm_userdispersing | Success | 20.40 | test_deploy_vms_with_varied_deploymentplanners.py
test_deployvm_userconcentrated | Success | 30.45 | test_deploy_vms_with_varied_deploymentplanners.py
test_deployvm_firstfit | Success | 50.45 | test_deploy_vms_with_varied_deploymentplanners.py
test_deployvm_userdata_post | Success | 20.34 | test_deploy_vm_with_userdata.py
test_deployvm_userdata | Success | 50.53 | test_deploy_vm_with_userdata.py
test_02_deploy_vm_root_resize | Success | 6.00 | test_deploy_vm_root_resize.py
test_01_deploy_vm_root_resize | Success | 5.70 | test_deploy_vm_root_resize.py
test_00_deploy_vm_root_resize | Success | 206.91 | test_deploy_vm_root_resize.py
test_deploy_vm_from_iso | Success | 207.03 | test_deploy_vm_iso.py
test_DeployVmAntiAffinityGroup | Success | 66.01 | test_affinity_groups.py
test_change_service_offering_for_vm_with_snapshots | Skipped | 0.00 | test_vm_snapshots.py
test_01_test_vm_volume_snapshot | Skipped | 0.00 | test_vm_snapshots.py
test_06_copy_template | Skipped | 0.00 | test_templates.py
test_static_role_account_acls | Skipped | 0.02 | test_staticroles.py
test_11_ss_nfs_version_on_ssvm | Skipped | 0.02 | test_ssvm.py
test_01_scale_vm | Skipped | 0.00 | test_scale_vm.py
test_01_primary_storage_iscsi | Skipped | 0.03 | test_primary_storage.py
test_nested_virtualization_vmware | Skipped | 0.00 | test_nested_virtualization.py
test_06_copy_iso | Skipped | 0.00 | test_iso.py
test_deploy_vgpu_enabled_vm | Skipped | 0.02 | test_deploy_vgpu_enabled_vm.py
test_3d_gpu_support | Skipped | 0.03 | test_deploy_vgpu_enabled_vm.py

borisstoyanov commented:
@nathanejohnson Thanks, for this good enhancement. Can we have a marvin tests that covers the new functionality?

wido commented on plugins/hypervisors/kvm/src/com/cloud/hypervisor/kvm/resource/LibvirtComputingResource.java:
```diff
@@ -2059,6 +2060,13 @@ So if getMinSpeed() returns null we fall back to getSpeed().
+        if (getGuestDiskModel(vmTO.getPlatformEmulator()) == DiskDef.DiskBus.SCSI) {
+            final SCSIDef sd = new SCSIDef((short)0, 0, 0, 9, 0);
+            devices.addDevice(sd);
+            s_logger.debug("Adding scsi definition:\n" + sd.toString());
```
Might want to include the name of the VM here as well? And SCSI is written in upper-case.

wido commented on plugins/hypervisors/kvm/src/com/cloud/hypervisor/kvm/resource/LibvirtComputingResource.java:
```diff
@@ -2158,6 +2166,7 @@ public int compare(final DiskTO arg0, final DiskTO arg1) {
 
             if (diskBusType == null) {
                 diskBusType = getGuestDiskModel(vmSpec.getPlatformEmulator());
+                s_logger.debug("disk bus type derived from getPlatformEmulator: " + vmSpec.getPlatformEmulator() + ", diskbustype is: "+diskBusType.toString());
```
Same here, please add the name of the VM

wido commented on plugins/hypervisors/kvm/src/com/cloud/hypervisor/kvm/resource/LibvirtComputingResource.java:
```diff
@@ -2216,6 +2225,8 @@ public int compare(final DiskTO arg0, final DiskTO arg1) {
                 }
             }
+
+            s_logger.debug("Adding disk: " + disk.toString());
```
Same here, please include the name of the VM

Also, do we really need all these logging lines?

nathanejohnson commented:
As an update, I'm working on a Marvin test, and I'm also planning to address wido's concerns.  Today had many distractions unfortunately.

borisstoyanov commented:
@nathanejohnson thank you, let me know when you're done with the changes so I could kick in smoke tests.

nathanejohnson commented:
@wido I have removed the extraneous log messages.

@boris I have added a smoke test for this that tests both virsh output from the host and lspci / lsblk output from the guest.  Here is the output from me running this in a bubble:

Test that libvirt properly created domain with scsi controller ... === TestName: test_01_verify_libvirt | Status : SUCCESS ===
ok
Verify that libvirt settings are as expected after a VM stop / start ... === TestName: test_02_verify_libvirt_after_restart | Status : SUCCESS ===
ok
Verify that libvirt settings are expected after a disk add ... === TestName: test_03_verify_libvirt_attach_disk | Status : SUCCESS ===
ok
Verify that guest sees scsi controller and disks ... === TestName: test_04_verify_guest_lspci | Status : SUCCESS ===
ok

----------------------------------------------------------------------
Ran 4 tests in 658.770s

OK

borisstoyanov commented:
Thanks @nathanejohnson, lets run that with BO
@blueorangutan package

blueorangutan commented:
@borisstoyanov a Jenkins job has been kicked to build packages. I'll keep you posted as I make progress.

blueorangutan commented:
Packaging result: ✔centos6 ✔centos7 ✔debian. JID-552

borisstoyanov commented:
@blueorangutan test

blueorangutan commented:
@borisstoyanov a Trillian-Jenkins test job (centos7 mgmt + kvm-centos7) has been kicked to run smoke tests

nathanejohnson commented:
@borisstoyanov Quick questions, what tags need to be present to get picked up by BO?  I'm worried my smoke test might get skipped.

blueorangutan commented:
<b>Trillian test result (tid-921)</b>
Environment: kvm-centos7 (x2), Advanced Networking with Mgmt server 7
Total time taken: 36820 seconds
Marvin logs: https://github.com/blueorangutan/acs-prs/releases/download/trillian/pr1955-t921-kvm-centos7.zip
Intermitten failure detected: /marvin/tests/smoke/test_deploy_virtio_scsi_vm.py
Intermitten failure detected: /marvin/tests/smoke/test_privategw_acl.py
Intermitten failure detected: /marvin/tests/smoke/test_snapshots.py
Intermitten failure detected: /marvin/tests/smoke/test_vpc_vpn.py
Test completed. 47 look ok, 3 have error(s)


Test | Result | Time (s) | Test File
--- | --- | --- | ---
test_04_rvpc_privategw_static_routes | `Failure` | 334.38 | test_privategw_acl.py
test_02_list_snapshots_with_removed_data_store | `Error` | 0.03 | test_snapshots.py
test_03_verify_libvirt_attach_disk | `Error` | 610.88 | test_deploy_virtio_scsi_vm.py
test_02_verify_libvirt_after_restart | `Error` | 615.96 | test_deploy_virtio_scsi_vm.py
test_01_verify_libvirt | `Error` | 600.73 | test_deploy_virtio_scsi_vm.py
test_01_vpc_site2site_vpn | Success | 159.71 | test_vpc_vpn.py
test_01_vpc_remote_access_vpn | Success | 65.83 | test_vpc_vpn.py
test_01_redundant_vpc_site2site_vpn | Success | 240.43 | test_vpc_vpn.py
test_02_VPC_default_routes | Success | 292.11 | test_vpc_router_nics.py
test_01_VPC_nics_after_destroy | Success | 551.36 | test_vpc_router_nics.py
test_05_rvpc_multi_tiers | Success | 514.33 | test_vpc_redundant.py
test_04_rvpc_network_garbage_collector_nics | Success | 1404.34 | test_vpc_redundant.py
test_03_create_redundant_VPC_1tier_2VMs_2IPs_2PF_ACL_reboot_routers | Success | 568.61 | test_vpc_redundant.py
test_02_redundant_VPC_default_routes | Success | 737.39 | test_vpc_redundant.py
test_01_create_redundant_VPC_2tiers_4VMs_4IPs_4PF_ACL | Success | 1271.92 | test_vpc_redundant.py
test_09_delete_detached_volume | Success | 156.17 | test_volumes.py
test_08_resize_volume | Success | 156.09 | test_volumes.py
test_07_resize_fail | Success | 156.08 | test_volumes.py
test_06_download_detached_volume | Success | 156.01 | test_volumes.py
test_05_detach_volume | Success | 150.53 | test_volumes.py
test_04_delete_attached_volume | Success | 150.91 | test_volumes.py
test_03_download_attached_volume | Success | 155.98 | test_volumes.py
test_02_attach_volume | Success | 94.76 | test_volumes.py
test_01_create_volume | Success | 711.91 | test_volumes.py
test_03_delete_vm_snapshots | Success | 275.10 | test_vm_snapshots.py
test_02_revert_vm_snapshots | Success | 100.69 | test_vm_snapshots.py
test_01_create_vm_snapshots | Success | 163.74 | test_vm_snapshots.py
test_deploy_vm_multiple | Success | 256.93 | test_vm_life_cycle.py
test_deploy_vm | Success | 0.02 | test_vm_life_cycle.py
test_advZoneVirtualRouter | Success | 0.02 | test_vm_life_cycle.py
test_10_attachAndDetach_iso | Success | 26.49 | test_vm_life_cycle.py
test_09_expunge_vm | Success | 125.12 | test_vm_life_cycle.py
test_08_migrate_vm | Success | 40.64 | test_vm_life_cycle.py
test_07_restore_vm | Success | 0.09 | test_vm_life_cycle.py
test_06_destroy_vm | Success | 125.64 | test_vm_life_cycle.py
test_03_reboot_vm | Success | 125.79 | test_vm_life_cycle.py
test_02_start_vm | Success | 10.12 | test_vm_life_cycle.py
test_01_stop_vm | Success | 40.25 | test_vm_life_cycle.py
test_CreateTemplateWithDuplicateName | Success | 35.32 | test_templates.py
test_08_list_system_templates | Success | 0.02 | test_templates.py
test_07_list_public_templates | Success | 0.03 | test_templates.py
test_05_template_permissions | Success | 0.04 | test_templates.py
test_04_extract_template | Success | 5.11 | test_templates.py
test_03_delete_template | Success | 5.09 | test_templates.py
test_02_edit_template | Success | 90.09 | test_templates.py
test_01_create_template | Success | 45.35 | test_templates.py
test_10_destroy_cpvm | Success | 161.57 | test_ssvm.py
test_09_destroy_ssvm | Success | 133.41 | test_ssvm.py
test_08_reboot_cpvm | Success | 101.46 | test_ssvm.py
test_07_reboot_ssvm | Success | 133.47 | test_ssvm.py
test_06_stop_cpvm | Success | 131.58 | test_ssvm.py
test_05_stop_ssvm | Success | 133.56 | test_ssvm.py
test_04_cpvm_internals | Success | 1.17 | test_ssvm.py
test_03_ssvm_internals | Success | 3.57 | test_ssvm.py
test_02_list_cpvm_vm | Success | 0.08 | test_ssvm.py
test_01_list_sec_storage_vm | Success | 0.09 | test_ssvm.py
test_01_snapshot_root_disk | Success | 10.92 | test_snapshots.py
test_04_change_offering_small | Success | 234.31 | test_service_offerings.py
test_03_delete_service_offering | Success | 0.03 | test_service_offerings.py
test_02_edit_service_offering | Success | 0.04 | test_service_offerings.py
test_01_create_service_offering | Success | 0.08 | test_service_offerings.py
test_02_sys_template_ready | Success | 0.08 | test_secondary_storage.py
test_01_sys_vm_start | Success | 0.12 | test_secondary_storage.py
test_09_reboot_router | Success | 40.26 | test_routers.py
test_08_start_router | Success | 35.24 | test_routers.py
test_07_stop_router | Success | 10.12 | test_routers.py
test_06_router_advanced | Success | 0.05 | test_routers.py
test_05_router_basic | Success | 0.03 | test_routers.py
test_04_restart_network_wo_cleanup | Success | 5.68 | test_routers.py
test_03_restart_network_cleanup | Success | 65.41 | test_routers.py
test_02_router_internal_adv | Success | 1.02 | test_routers.py
test_01_router_internal_basic | Success | 0.56 | test_routers.py
test_router_dns_guestipquery | Success | 76.63 | test_router_dns.py
test_router_dns_externalipquery | Success | 0.06 | test_router_dns.py
test_router_dhcphosts | Success | 276.46 | test_router_dhcphosts.py
test_router_dhcp_opts | Success | 21.51 | test_router_dhcphosts.py
test_01_updatevolumedetail | Success | 0.07 | test_resource_detail.py
test_01_reset_vm_on_reboot | Success | 145.77 | test_reset_vm_on_reboot.py
test_createRegion | Success | 0.03 | test_regions.py
test_create_pvlan_network | Success | 5.15 | test_pvlan.py
test_dedicatePublicIpRange | Success | 0.31 | test_public_ip_range.py
test_03_vpc_privategw_restart_vpc_cleanup | Success | 478.64 | test_privategw_acl.py
test_02_vpc_privategw_static_routes | Success | 353.91 | test_privategw_acl.py
test_01_vpc_privategw_acl | Success | 86.54 | test_privategw_acl.py
test_01_primary_storage_nfs | Success | 35.92 | test_primary_storage.py
test_createPortablePublicIPRange | Success | 15.13 | test_portable_publicip.py
test_createPortablePublicIPAcquire | Success | 15.36 | test_portable_publicip.py
test_isolate_network_password_server | Success | 89.23 | test_password_server.py
test_UpdateStorageOverProvisioningFactor | Success | 0.10 | test_over_provisioning.py
test_oobm_zchange_password | Success | 25.49 | test_outofbandmanagement.py
test_oobm_multiple_mgmt_server_ownership | Success | 16.24 | test_outofbandmanagement.py
test_oobm_issue_power_status | Success | 10.17 | test_outofbandmanagement.py
test_oobm_issue_power_soft | Success | 15.25 | test_outofbandmanagement.py
test_oobm_issue_power_reset | Success | 15.25 | test_outofbandmanagement.py
test_oobm_issue_power_on | Success | 15.23 | test_outofbandmanagement.py
test_oobm_issue_power_off | Success | 15.28 | test_outofbandmanagement.py
test_oobm_issue_power_cycle | Success | 15.26 | test_outofbandmanagement.py
test_oobm_enabledisable_across_clusterzones | Success | 87.15 | test_outofbandmanagement.py
test_oobm_enable_feature_valid | Success | 5.12 | test_outofbandmanagement.py
test_oobm_enable_feature_invalid | Success | 0.07 | test_outofbandmanagement.py
test_oobm_disable_feature_valid | Success | 5.13 | test_outofbandmanagement.py
test_oobm_disable_feature_invalid | Success | 0.07 | test_outofbandmanagement.py
test_oobm_configure_invalid_driver | Success | 0.06 | test_outofbandmanagement.py
test_oobm_configure_default_driver | Success | 0.06 | test_outofbandmanagement.py
test_oobm_background_powerstate_sync | Success | 23.29 | test_outofbandmanagement.py
test_extendPhysicalNetworkVlan | Success | 15.23 | test_non_contigiousvlan.py
test_01_nic | Success | 433.75 | test_nic.py
test_releaseIP | Success | 242.55 | test_network.py
test_reboot_router | Success | 428.00 | test_network.py
test_public_ip_user_account | Success | 10.18 | test_network.py
test_public_ip_admin_account | Success | 40.21 | test_network.py
test_network_rules_acquired_public_ip_3_Load_Balancer_Rule | Success | 66.63 | test_network.py
test_network_rules_acquired_public_ip_2_nat_rule | Success | 61.63 | test_network.py
test_network_rules_acquired_public_ip_1_static_nat_rule | Success | 123.62 | test_network.py
test_delete_account | Success | 282.09 | test_network.py
test_02_port_fwd_on_non_src_nat | Success | 55.48 | test_network.py
test_01_port_fwd_on_src_nat | Success | 111.60 | test_network.py
test_nic_secondaryip_add_remove | Success | 232.38 | test_multipleips_per_nic.py
login_test_saml_user | Success | 17.75 | test_login.py
test_assign_and_removal_lb | Success | 133.26 | test_loadbalance.py
test_02_create_lb_rule_non_nat | Success | 187.13 | test_loadbalance.py
test_01_create_lb_rule_src_nat | Success | 217.50 | test_loadbalance.py
test_03_list_snapshots | Success | 0.06 | test_list_ids_parameter.py
test_02_list_templates | Success | 0.03 | test_list_ids_parameter.py
test_01_list_volumes | Success | 0.02 | test_list_ids_parameter.py
test_07_list_default_iso | Success | 0.04 | test_iso.py
test_05_iso_permissions | Success | 0.04 | test_iso.py
test_04_extract_Iso | Success | 5.13 | test_iso.py
test_03_delete_iso | Success | 95.17 | test_iso.py
test_02_edit_iso | Success | 0.06 | test_iso.py
test_01_create_iso | Success | 20.69 | test_iso.py
test_04_rvpc_internallb_haproxy_stats_on_all_interfaces | Success | 192.59 | test_internal_lb.py
test_03_vpc_internallb_haproxy_stats_on_all_interfaces | Success | 146.95 | test_internal_lb.py
test_02_internallb_roundrobin_1RVPC_3VM_HTTP_port80 | Success | 499.91 | test_internal_lb.py
test_01_internallb_roundrobin_1VPC_3VM_HTTP_port80 | Success | 424.37 | test_internal_lb.py
test_dedicateGuestVlanRange | Success | 10.20 | test_guest_vlan_range.py
test_UpdateConfigParamWithScope | Success | 0.09 | test_global_settings.py
test_rolepermission_lifecycle_update | Success | 5.78 | test_dynamicroles.py
test_rolepermission_lifecycle_list | Success | 5.68 | test_dynamicroles.py
test_rolepermission_lifecycle_delete | Success | 5.57 | test_dynamicroles.py
test_rolepermission_lifecycle_create | Success | 5.59 | test_dynamicroles.py
test_rolepermission_lifecycle_concurrent_updates | Success | 5.67 | test_dynamicroles.py
test_role_lifecycle_update_role_inuse | Success | 5.62 | test_dynamicroles.py
test_role_lifecycle_update | Success | 10.65 | test_dynamicroles.py
test_role_lifecycle_list | Success | 5.63 | test_dynamicroles.py
test_role_lifecycle_delete | Success | 10.62 | test_dynamicroles.py
test_role_lifecycle_create | Success | 5.59 | test_dynamicroles.py
test_role_inuse_deletion | Success | 5.58 | test_dynamicroles.py
test_role_account_acls_multiple_mgmt_servers | Success | 6.88 | test_dynamicroles.py
test_role_account_acls | Success | 7.02 | test_dynamicroles.py
test_default_role_deletion | Success | 5.64 | test_dynamicroles.py
test_04_create_fat_type_disk_offering | Success | 0.05 | test_disk_offerings.py
test_03_delete_disk_offering | Success | 0.03 | test_disk_offerings.py
test_02_edit_disk_offering | Success | 0.03 | test_disk_offerings.py
test_02_create_sparse_type_disk_offering | Success | 0.05 | test_disk_offerings.py
test_01_create_disk_offering | Success | 0.07 | test_disk_offerings.py
test_deployvm_userdispersing | Success | 20.40 | test_deploy_vms_with_varied_deploymentplanners.py
test_deployvm_userconcentrated | Success | 20.39 | test_deploy_vms_with_varied_deploymentplanners.py
test_deployvm_firstfit | Success | 85.58 | test_deploy_vms_with_varied_deploymentplanners.py
test_deployvm_userdata_post | Success | 10.27 | test_deploy_vm_with_userdata.py
test_deployvm_userdata | Success | 65.55 | test_deploy_vm_with_userdata.py
test_02_deploy_vm_root_resize | Success | 5.64 | test_deploy_vm_root_resize.py
test_01_deploy_vm_root_resize | Success | 5.65 | test_deploy_vm_root_resize.py
test_00_deploy_vm_root_resize | Success | 226.78 | test_deploy_vm_root_resize.py
test_deploy_vm_from_iso | Success | 211.76 | test_deploy_vm_iso.py
test_04_verify_guest_lspci | Success | 1.03 | test_deploy_virtio_scsi_vm.py
test_DeployVmAntiAffinityGroup | Success | 65.65 | test_affinity_groups.py
test_change_service_offering_for_vm_with_snapshots | Skipped | 0.00 | test_vm_snapshots.py
test_01_test_vm_volume_snapshot | Skipped | 0.00 | test_vm_snapshots.py
test_06_copy_template | Skipped | 0.00 | test_templates.py
test_static_role_account_acls | Skipped | 0.02 | test_staticroles.py
test_11_ss_nfs_version_on_ssvm | Skipped | 0.02 | test_ssvm.py
test_01_scale_vm | Skipped | 0.00 | test_scale_vm.py
test_01_primary_storage_iscsi | Skipped | 0.03 | test_primary_storage.py
test_nested_virtualization_vmware | Skipped | 0.00 | test_nested_virtualization.py
test_06_copy_iso | Skipped | 0.00 | test_iso.py
test_deploy_vgpu_enabled_vm | Skipped | 0.02 | test_deploy_vgpu_enabled_vm.py
test_3d_gpu_support | Skipped | 0.03 | test_deploy_vgpu_enabled_vm.py

nathanejohnson commented:
@borisstoyanov so the good news is it didn't skip the test.  the bad news is it didn't seem to have proper credentials for ssh'ing into your hosts.  It did pass the test where it ssh'ed into the guest, however.  So this at least demonstrates a guest is happy.

Do you have any tips on ways of making the smoke test play nice with BO wrt ssh'ing into the kvm hosts?

borisstoyanov commented:
Thanks @nathanejohnson I will have a look

remibergsma commented:
Tested it manually, works great. LGTM.

borisstoyanov commented:
I think that's the issue at our environment, it's trying to ssh via hostname instead of IP. Can you change that to IP please @nathanejohnson ? 
`====Trying SSH Connection: Host:pr1955-t929-kvm-centos6-kvm1 User:root                                   Port:22 RetryCnt:3===`

nathanejohnson commented:
@borisstoyanov I have reworked the test so that it should now use the host IP versus the hostname.  Could you please re-run the blue orangutan test?  Thank you!

nathanejohnson commented:
Also, I just made a commit, I have not squashed it because I wanted to get some feedback.  I noticed that vmware uses details to pass rootDiskController for the bus type, and there was even some code that inspected this in the kvm plugin.  With this latest commit, it will actually honor either OS type "Other PV Virtio-SCSI" or rootDiskController passed in via details, with details overriding OS type.  I also made a minor tweak to the UI to allow setting the root disk bus via the add template dialog in the UI:

![template_screen](https://cloud.githubusercontent.com/assets/1360240/23617743/b29b8dd6-0253-11e7-970b-6aafe314bd49.png)

So this way you can select virtio scsi by one of the following three ways:

1) selecting os type as "Other PV Virtio-SCSI (64 bit)"
2) uploading a template and specifying details[0].rootDiskController=scsi, either from the UI or the API, then creating a VM from this template
3) deploying a VM and specifying details[0].rootDiskController=scsi , currently only available via the API

nathanejohnson commented:
@borisstoyanov Whenever you get a second could you kick off a blue orangutan smoke test?  I believe I have fixed the issue with ssh'ing into the KVM hosts, I sue the IP now instead of hostname.

borisstoyanov commented:
@nathanejohnson sure, sorry I've missed your comment about the test fix earlier. 
@blueorangutan package

blueorangutan commented:
@borisstoyanov a Jenkins job has been kicked to build packages. I'll keep you posted as I make progress.

blueorangutan commented:
Packaging result: ✔centos6 ✔centos7 ✔debian. JID-577

borisstoyanov commented:
@blueorangutan test

blueorangutan commented:
@borisstoyanov a Trillian-Jenkins test job (centos7 mgmt + kvm-centos7) has been kicked to run smoke tests

blueorangutan commented:
<b>Trillian test result (tid-938)</b>
Environment: kvm-centos7 (x2), Advanced Networking with Mgmt server 7
Total time taken: 34569 seconds
Marvin logs: https://github.com/blueorangutan/acs-prs/releases/download/trillian/pr1955-t938-kvm-centos7.zip
Intermitten failure detected: /marvin/tests/smoke/test_privategw_acl.py
Intermitten failure detected: /marvin/tests/smoke/test_snapshots.py
Intermitten failure detected: /marvin/tests/smoke/test_vpc_redundant.py
Test completed. 47 look ok, 3 have error(s)


Test | Result | Time (s) | Test File
--- | --- | --- | ---
test_02_redundant_VPC_default_routes | `Failure` | 879.31 | test_vpc_redundant.py
test_04_rvpc_privategw_static_routes | `Failure` | 362.02 | test_privategw_acl.py
test_03_vpc_privategw_restart_vpc_cleanup | `Failure` | 873.50 | test_privategw_acl.py
test_02_list_snapshots_with_removed_data_store | `Error` | 0.04 | test_snapshots.py
test_01_vpc_site2site_vpn | Success | 170.70 | test_vpc_vpn.py
test_01_vpc_remote_access_vpn | Success | 66.07 | test_vpc_vpn.py
test_01_redundant_vpc_site2site_vpn | Success | 266.31 | test_vpc_vpn.py
test_02_VPC_default_routes | Success | 297.40 | test_vpc_router_nics.py
test_01_VPC_nics_after_destroy | Success | 543.52 | test_vpc_router_nics.py
test_05_rvpc_multi_tiers | Success | 521.44 | test_vpc_redundant.py
test_04_rvpc_network_garbage_collector_nics | Success | 1421.18 | test_vpc_redundant.py
test_03_create_redundant_VPC_1tier_2VMs_2IPs_2PF_ACL_reboot_routers | Success | 569.99 | test_vpc_redundant.py
test_01_create_redundant_VPC_2tiers_4VMs_4IPs_4PF_ACL | Success | 1292.16 | test_vpc_redundant.py
test_09_delete_detached_volume | Success | 156.42 | test_volumes.py
test_08_resize_volume | Success | 156.48 | test_volumes.py
test_07_resize_fail | Success | 161.51 | test_volumes.py
test_06_download_detached_volume | Success | 156.29 | test_volumes.py
test_05_detach_volume | Success | 155.77 | test_volumes.py
test_04_delete_attached_volume | Success | 146.33 | test_volumes.py
test_03_download_attached_volume | Success | 156.26 | test_volumes.py
test_02_attach_volume | Success | 95.12 | test_volumes.py
test_01_create_volume | Success | 714.40 | test_volumes.py
test_03_delete_vm_snapshots | Success | 275.21 | test_vm_snapshots.py
test_02_revert_vm_snapshots | Success | 100.75 | test_vm_snapshots.py
test_01_create_vm_snapshots | Success | 163.76 | test_vm_snapshots.py
test_deploy_vm_multiple | Success | 282.81 | test_vm_life_cycle.py
test_deploy_vm | Success | 0.03 | test_vm_life_cycle.py
test_advZoneVirtualRouter | Success | 0.02 | test_vm_life_cycle.py
test_10_attachAndDetach_iso | Success | 26.68 | test_vm_life_cycle.py
test_09_expunge_vm | Success | 125.22 | test_vm_life_cycle.py
test_08_migrate_vm | Success | 30.86 | test_vm_life_cycle.py
test_07_restore_vm | Success | 0.16 | test_vm_life_cycle.py
test_06_destroy_vm | Success | 125.79 | test_vm_life_cycle.py
test_03_reboot_vm | Success | 125.81 | test_vm_life_cycle.py
test_02_start_vm | Success | 10.22 | test_vm_life_cycle.py
test_01_stop_vm | Success | 40.33 | test_vm_life_cycle.py
test_CreateTemplateWithDuplicateName | Success | 40.50 | test_templates.py
test_08_list_system_templates | Success | 0.03 | test_templates.py
test_07_list_public_templates | Success | 0.04 | test_templates.py
test_05_template_permissions | Success | 0.06 | test_templates.py
test_04_extract_template | Success | 5.14 | test_templates.py
test_03_delete_template | Success | 5.11 | test_templates.py
test_02_edit_template | Success | 90.21 | test_templates.py
test_01_create_template | Success | 25.35 | test_templates.py
test_10_destroy_cpvm | Success | 161.80 | test_ssvm.py
test_09_destroy_ssvm | Success | 193.76 | test_ssvm.py
test_08_reboot_cpvm | Success | 131.68 | test_ssvm.py
test_07_reboot_ssvm | Success | 133.61 | test_ssvm.py
test_06_stop_cpvm | Success | 131.75 | test_ssvm.py
test_05_stop_ssvm | Success | 133.78 | test_ssvm.py
test_04_cpvm_internals | Success | 1.29 | test_ssvm.py
test_03_ssvm_internals | Success | 3.64 | test_ssvm.py
test_02_list_cpvm_vm | Success | 0.12 | test_ssvm.py
test_01_list_sec_storage_vm | Success | 0.12 | test_ssvm.py
test_01_snapshot_root_disk | Success | 11.19 | test_snapshots.py
test_04_change_offering_small | Success | 239.57 | test_service_offerings.py
test_03_delete_service_offering | Success | 0.04 | test_service_offerings.py
test_02_edit_service_offering | Success | 0.05 | test_service_offerings.py
test_01_create_service_offering | Success | 0.11 | test_service_offerings.py
test_02_sys_template_ready | Success | 0.13 | test_secondary_storage.py
test_01_sys_vm_start | Success | 0.18 | test_secondary_storage.py
test_09_reboot_router | Success | 35.31 | test_routers.py
test_08_start_router | Success | 35.36 | test_routers.py
test_07_stop_router | Success | 10.21 | test_routers.py
test_06_router_advanced | Success | 0.06 | test_routers.py
test_05_router_basic | Success | 0.04 | test_routers.py
test_04_restart_network_wo_cleanup | Success | 5.73 | test_routers.py
test_03_restart_network_cleanup | Success | 65.81 | test_routers.py
test_02_router_internal_adv | Success | 1.06 | test_routers.py
test_01_router_internal_basic | Success | 0.57 | test_routers.py
test_router_dns_guestipquery | Success | 76.77 | test_router_dns.py
test_router_dns_externalipquery | Success | 0.08 | test_router_dns.py
test_router_dhcphosts | Success | 279.84 | test_router_dhcphosts.py
test_router_dhcp_opts | Success | 22.28 | test_router_dhcphosts.py
test_01_updatevolumedetail | Success | 0.10 | test_resource_detail.py
test_01_reset_vm_on_reboot | Success | 146.14 | test_reset_vm_on_reboot.py
test_createRegion | Success | 0.04 | test_regions.py
test_create_pvlan_network | Success | 5.22 | test_pvlan.py
test_dedicatePublicIpRange | Success | 0.51 | test_public_ip_range.py
test_02_vpc_privategw_static_routes | Success | 375.67 | test_privategw_acl.py
test_01_vpc_privategw_acl | Success | 92.31 | test_privategw_acl.py
test_01_primary_storage_nfs | Success | 35.81 | test_primary_storage.py
test_createPortablePublicIPRange | Success | 15.20 | test_portable_publicip.py
test_createPortablePublicIPAcquire | Success | 15.48 | test_portable_publicip.py
test_isolate_network_password_server | Success | 89.91 | test_password_server.py
test_UpdateStorageOverProvisioningFactor | Success | 0.13 | test_over_provisioning.py
test_oobm_zchange_password | Success | 30.63 | test_outofbandmanagement.py
test_oobm_multiple_mgmt_server_ownership | Success | 16.32 | test_outofbandmanagement.py
test_oobm_issue_power_status | Success | 10.23 | test_outofbandmanagement.py
test_oobm_issue_power_soft | Success | 15.30 | test_outofbandmanagement.py
test_oobm_issue_power_reset | Success | 15.31 | test_outofbandmanagement.py
test_oobm_issue_power_on | Success | 15.33 | test_outofbandmanagement.py
test_oobm_issue_power_off | Success | 15.42 | test_outofbandmanagement.py
test_oobm_issue_power_cycle | Success | 10.49 | test_outofbandmanagement.py
test_oobm_enabledisable_across_clusterzones | Success | 87.55 | test_outofbandmanagement.py
test_oobm_enable_feature_valid | Success | 5.15 | test_outofbandmanagement.py
test_oobm_enable_feature_invalid | Success | 0.10 | test_outofbandmanagement.py
test_oobm_disable_feature_valid | Success | 5.18 | test_outofbandmanagement.py
test_oobm_disable_feature_invalid | Success | 0.14 | test_outofbandmanagement.py
test_oobm_configure_invalid_driver | Success | 0.08 | test_outofbandmanagement.py
test_oobm_configure_default_driver | Success | 0.07 | test_outofbandmanagement.py
test_oobm_background_powerstate_sync | Success | 29.43 | test_outofbandmanagement.py
test_extendPhysicalNetworkVlan | Success | 15.31 | test_non_contigiousvlan.py
test_01_nic | Success | 424.12 | test_nic.py
test_releaseIP | Success | 268.17 | test_network.py
test_reboot_router | Success | 414.00 | test_network.py
test_public_ip_user_account | Success | 10.27 | test_network.py
test_public_ip_admin_account | Success | 40.27 | test_network.py
test_network_rules_acquired_public_ip_3_Load_Balancer_Rule | Success | 66.95 | test_network.py
test_network_rules_acquired_public_ip_2_nat_rule | Success | 61.91 | test_network.py
test_network_rules_acquired_public_ip_1_static_nat_rule | Success | 123.89 | test_network.py
test_delete_account | Success | 272.64 | test_network.py
test_02_port_fwd_on_non_src_nat | Success | 55.62 | test_network.py
test_01_port_fwd_on_src_nat | Success | 111.75 | test_network.py
test_nic_secondaryip_add_remove | Success | 227.64 | test_multipleips_per_nic.py
login_test_saml_user | Success | 19.14 | test_login.py
test_assign_and_removal_lb | Success | 133.70 | test_loadbalance.py
test_02_create_lb_rule_non_nat | Success | 187.27 | test_loadbalance.py
test_01_create_lb_rule_src_nat | Success | 218.79 | test_loadbalance.py
test_03_list_snapshots | Success | 0.06 | test_list_ids_parameter.py
test_02_list_templates | Success | 0.04 | test_list_ids_parameter.py
test_01_list_volumes | Success | 0.03 | test_list_ids_parameter.py
test_07_list_default_iso | Success | 0.06 | test_iso.py
test_05_iso_permissions | Success | 0.06 | test_iso.py
test_04_extract_Iso | Success | 5.15 | test_iso.py
test_03_delete_iso | Success | 95.18 | test_iso.py
test_02_edit_iso | Success | 0.08 | test_iso.py
test_01_create_iso | Success | 20.99 | test_iso.py
test_04_rvpc_internallb_haproxy_stats_on_all_interfaces | Success | 203.35 | test_internal_lb.py
test_03_vpc_internallb_haproxy_stats_on_all_interfaces | Success | 148.59 | test_internal_lb.py
test_02_internallb_roundrobin_1RVPC_3VM_HTTP_port80 | Success | 512.36 | test_internal_lb.py
test_01_internallb_roundrobin_1VPC_3VM_HTTP_port80 | Success | 429.49 | test_internal_lb.py
test_dedicateGuestVlanRange | Success | 10.29 | test_guest_vlan_range.py
test_UpdateConfigParamWithScope | Success | 0.14 | test_global_settings.py
test_rolepermission_lifecycle_update | Success | 6.15 | test_dynamicroles.py
test_rolepermission_lifecycle_list | Success | 5.95 | test_dynamicroles.py
test_rolepermission_lifecycle_delete | Success | 5.83 | test_dynamicroles.py
test_rolepermission_lifecycle_create | Success | 6.03 | test_dynamicroles.py
test_rolepermission_lifecycle_concurrent_updates | Success | 6.05 | test_dynamicroles.py
test_role_lifecycle_update_role_inuse | Success | 6.29 | test_dynamicroles.py
test_role_lifecycle_update | Success | 10.94 | test_dynamicroles.py
test_role_lifecycle_list | Success | 5.87 | test_dynamicroles.py
test_role_lifecycle_delete | Success | 5.88 | test_dynamicroles.py
test_role_lifecycle_create | Success | 5.88 | test_dynamicroles.py
test_role_inuse_deletion | Success | 5.87 | test_dynamicroles.py
test_role_account_acls_multiple_mgmt_servers | Success | 7.96 | test_dynamicroles.py
test_role_account_acls | Success | 8.17 | test_dynamicroles.py
test_default_role_deletion | Success | 5.99 | test_dynamicroles.py
test_04_create_fat_type_disk_offering | Success | 0.07 | test_disk_offerings.py
test_03_delete_disk_offering | Success | 0.04 | test_disk_offerings.py
test_02_edit_disk_offering | Success | 0.05 | test_disk_offerings.py
test_02_create_sparse_type_disk_offering | Success | 0.07 | test_disk_offerings.py
test_01_create_disk_offering | Success | 0.24 | test_disk_offerings.py
test_deployvm_userdispersing | Success | 20.55 | test_deploy_vms_with_varied_deploymentplanners.py
test_deployvm_userconcentrated | Success | 20.57 | test_deploy_vms_with_varied_deploymentplanners.py
test_deployvm_firstfit | Success | 65.68 | test_deploy_vms_with_varied_deploymentplanners.py
test_deployvm_userdata_post | Success | 10.42 | test_deploy_vm_with_userdata.py
test_deployvm_userdata | Success | 70.87 | test_deploy_vm_with_userdata.py
test_02_deploy_vm_root_resize | Success | 5.95 | test_deploy_vm_root_resize.py
test_01_deploy_vm_root_resize | Success | 5.97 | test_deploy_vm_root_resize.py
test_00_deploy_vm_root_resize | Success | 223.00 | test_deploy_vm_root_resize.py
test_deploy_vm_from_iso | Success | 212.36 | test_deploy_vm_iso.py
test_06_verify_guest_lspci_again | Success | 40.76 | test_deploy_virtio_scsi_vm.py
test_05_change_vm_ostype_restart | Success | 21.01 | test_deploy_virtio_scsi_vm.py
test_04_verify_guest_lspci | Success | 40.89 | test_deploy_virtio_scsi_vm.py
test_03_verify_libvirt_attach_disk | Success | 10.82 | test_deploy_virtio_scsi_vm.py
test_02_verify_libvirt_after_restart | Success | 131.58 | test_deploy_virtio_scsi_vm.py
test_01_verify_libvirt | Success | 0.47 | test_deploy_virtio_scsi_vm.py
test_DeployVmAntiAffinityGroup | Success | 65.93 | test_affinity_groups.py
test_change_service_offering_for_vm_with_snapshots | Skipped | 0.00 | test_vm_snapshots.py
test_01_test_vm_volume_snapshot | Skipped | 0.00 | test_vm_snapshots.py
test_06_copy_template | Skipped | 0.00 | test_templates.py
test_static_role_account_acls | Skipped | 0.02 | test_staticroles.py
test_11_ss_nfs_version_on_ssvm | Skipped | 0.02 | test_ssvm.py
test_01_scale_vm | Skipped | 0.00 | test_scale_vm.py
test_01_primary_storage_iscsi | Skipped | 0.04 | test_primary_storage.py
test_nested_virtualization_vmware | Skipped | 0.00 | test_nested_virtualization.py
test_06_copy_iso | Skipped | 0.00 | test_iso.py
test_deploy_vgpu_enabled_vm | Skipped | 0.03 | test_deploy_vgpu_enabled_vm.py
test_3d_gpu_support | Skipped | 0.04 | test_deploy_vgpu_enabled_vm.py

kiwiflyer commented:
tag:mergeready

wido commented:
I would like to give a LGTM, but it's only the logging lines which I don't like yet.

They seem like personal debug lines, but a admin will never know what they actually mean.

In addition, I would like to see discard/unmap enabled by default for Virtio-SCSI disks: https://blog.widodh.nl/2015/12/using-trimdiscard-with-ceph-rbd-and-libvirt/

This will also allow for QCOW2 and RBD trimming.

nathanejohnson commented:
@wido I have removed all of the debug logging statements that didn't exist before I touched the code, I have also added the discard option to the disks.

wido commented:
Do we want to pass the unmap not as a string? Shouldn't that be a enum?

nathanejohnson commented:
@wido Actually I was just about to push a commit that changed the string to enum, but builds are failing locally because of this test:
    
    assertTrue(QuotaAlertManagerImpl.getDifferenceDays(now, new DateTime(now).plusDays(1).toDate()) == 1L);

Daylight savings time strikes again

wido commented:
@nathanejohnson: Good! I would like to see that being enums.

nathanejohnson commented:
I just re-depoyed and re-ran the test:

    Test that libvirt properly created domain with scsi controller ... === TestName: test_01_verify_libvirt | Status : SUCCESS ===
    ok
    Verify that libvirt settings are as expected after a VM stop / start ... === TestName: test_02_verify_libvirt_after_restart | Status : SUCCESS ===
    ok
    Verify that libvirt settings are expected after a disk add ... === TestName: test_03_verify_libvirt_attach_disk | Status : SUCCESS ===
    ok
    Verify that guest sees scsi controller and disks ... === TestName: test_04_verify_guest_lspci | Status : SUCCESS ===
    ok
    Update os type to Ubuntu, change vm details rootdiskController ... === TestName: test_05_change_vm_ostype_restart | Status : SUCCESS ===
    ok
    Verify that guest sees scsi controller and disks after switching ostype and rdc ... === TestName: test_06_verify_guest_lspci_again | Status : SUCCESS ===
    ok
    
    ----------------------------------------------------------------------
    Ran 6 tests in 785.914s
    
    OK

@wido , I believe all of your concerns should be addressed.  trim is enabled, it's using an enum and not a string, and spurious log messages have been eliminated.

I have squashed all of the commits into one.  I'm at a point where I'm happy with the code.

wido commented:
LGTM from the code

kiwiflyer commented:
@karuturi  3 x LGTM, testing successful. Ready for Merge.

wido commented:
Yes, we are indeed ready for a merge. Shall we do that?

karuturi commented:
merging..

DaanHoogland commented:
@nathanejohnson @wido @remibergsma @karuturi @kiwiflyer @dmambry
in #2044 tests are failing on vmware while this is only for kvm. Can we add a check on the tests for this or is it supposed to work for all hypervisors?

nathanejohnson commented:
@DaanHoogland @wido @remibergsma @karuturi @kiwiflyer I think I see the issue with the test, I should hopefully have a new PR opened later today.
